src/Sarif.Converters/FortifyFprConverter.cs

```diff
@@ -464,9 +464,10 @@ private void ParseVulnerability(IResultLogWriter output)
                     // Get the rule GUID from the ClassId element.
                     string ruleId = _reader.ReadElementContentAsString();
                     rule = FindOrCreateRule(ruleId, out ruleIndex);
-
+                    
                     result.RuleIndex = ruleIndex;
                     result.Level = GetFailureLevelFromRuleMetadata(rule);
+                    rule.DefaultConfiguration.Level = (FailureLevel)GetFailureLevelFromRuleMetadata(rule);
```

Comment:
>GetFailureLevelFromRuleMetadata [](start = 68, length = 31)

1. The return type of `GetFailureLevelFromRuleMetadata` is `FailureLevel`, so there is no need to cast.
2. Don't call `GetFailureLevelFromRuleMetadata` more than once. Call it once, capture its return value, and assign it to both places.
3. You are assigning to a property of `rule.DefaultConfiguration`, but `rule.DefaultConfiguration` is never assigned. `FindOrCreateRule` does not initialize it. I am surprised this does not crash.

Reply:
Sure, will do pint 1 and 2. 
For point 3 I have already added code to create DefaultConfiguration when Rule is being created.

Reply:
Oh, I see it now, thanks.

---
In reply to: [314902430](https://github.com/microsoft/sarif-sdk/pull/1650#discussion_r314902430) [](ancestors = 314902430)